### ISSUE TITLE
Fix Request: AvoidUnguardedMutableInheritedFieldsInSharedObjects false positive for lambda

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#496]
example:
```
import org.springframework.stereotype.Service;

@Service
public class Foo extends Bar {

    public void baz(List<Response> list, List<Order> orders) {
            list.stream().forEach(response1 -> response1.setOrders(orders)); // <- false positive;  note: misuse of forEach
    }
}
```
